### ISSUE TITLE
Ambient Sounds

### ISSUE THREAD

**Issue by edale2** [Insidious611/DancingMadFF6#32]
OK, I'm doing a runthrough to find sound effects that it might be safe to add without causing any conflicts.

I'm going to upload the completed PCM's I make for this to this thread, but there's a good chance at least some of them won't be usable, and once I know for sure a track is good, I might try and make a better one, since these are mostly for testing at this point.

I'll also have notes on where I am in the playthrough when I add the PCM, so I can do a second runthrough to try the earlier parts with that PCM included.

**Comment by edale2** [comment]
https://mega.nz/#!eJhD2QLD!nvDkzPAXHXBk5orTa15APP9N6Wm3gHTBAPsRwBd3wWw

Track 57
Narshe's wind sound.
Added right at the beginning, first use between Opening 2 and 3 (when biggs and wedge are discussing the "witch" they have with them).
Needs to be tested in all later areas involving Narshe.

**Comment by edale2** [comment]
Possible bug (only if track 57 is usable):
In Locke's scenario (occupied South Figaro, when you need to change Locke's outfit a few times), in the rich man's house (where you're supposed to listen for the drafty room). Once you go to the second floor (or enter/exit the menu on the first floor), track 57 plays at full volume, rather than muted (it's supposed to be barely audible in the 2nd floor hallway, and quiet but audible in the room with the secret passage, going to loud in the first room of the secret passage, before switching to music on the next room/start of mini-dungeon).

I imagine it's a matter of setting some volume handlers on those specific audio hooks, but not sure how complex that actually is.

It should probably be tested in another emulator as well (I'm using Snes9x), incase it's emulator specific.

SRM attached. Just give the code "Courage" to the boy to open the secret passage to get to the rich man's house.

[ff3.zip](https://github.com/Insidious611/DancingMadFF6/files/1393470/ff3.zip)

**Comment by edale2** [comment]
Had to check that one on an unmodded ROM to make sure there's not supposed to be any music playing in that part, almost thought I had to say the track's not usable, but I guess it's staying in for the rest of this playthrough.

I'm probably going to have to find a different sound clip if I use this though, current one is GREAT for Narshe (and I found a GREAT loop point), but not so good for an indoors wind, too deep a pitch.

**Comment by edale2** [comment]
Entering the room with the secret passage, then exiting to the hallway will put the wind sound at the correct volume for the hallway (still PCM audio). At least until you exit the menu screen.

**Comment by edale2** [comment]
SFX / Ambient Sounds location master list (to be updated as new ones are located... and probably reformatted when I'm not on 30+ hours without sleep, lol):

---
Track 30 - "waterfall": Lethe River (raft, WoB) / Barron Falls (Sabin jump, Sabin's scenario, WoB) / Cave after Veldt (where you jump into the Serpent Trench, Sabin's scebario, WoB)

Track 57 "wind": Narshe (beginning, WoB) / South Figaro, inside rich man's house (Locke's scenario, WoB) / The Sealed Gate (right before the Espers are released, WoB) / Mobilz (WoR)

Track 58 (waves): Solitary island (WoR) / immediately after Barron Falls (Sabin washing ashore, Sabin's scenario WoB) / Mobliz, Serpent Trench cutscene (when you talk to the guy and tell him you haven't heard about it, WoB) / On the ship from Nikeah to Narshe (WoB)

Track 74 (Fire/house burning): Thamasa house burning down (WoB)

---
Should make things easier both for me, and for anyone who wants to replace the SFX themselves down the line, to have this info condensed into one post.

**Comment by edale2** [comment]
God, it's a good thing my slowdown of that waves track to really sell the scene in Solitary Island still seems to work well in all these other places it's showing up, lol. Thankfully it's not playing with any music so far.

**Comment by Dizzy611** [comment]
Just wanted you to know that I'm watching your progress here but have not tested anything you've linked yet as I'm deep in my main testing pre-release. So uhh, don't think I'm ignoring this :P

**Comment by edale2** [comment]
I kinda figured that. This thread is more for my own tracking of things as well as notes for anyone doing work on this down the line.

**Comment by edale2** [comment]
https://mega.nz/#!aFxDFLjL!k2NChHGXyLjh23zVLSGFDqb9wa0iUiZs1_MEUJ__8II
House burning down effect.
Thamasa events when you meet Strago and Relm.

Decent track, but look for one that gives more of a "in a burning down house" feel to it.

**Comment by edale2** [comment]
Well it's a good thing I'm pretty much done with all the editing for the existing files, because I just lost my doc with my notes on where everything's loop points are... shit. I was doing so good keeping everything documented too...

**Comment by edale2** [comment]
And then I have a freeze start AS I'm saving, I thought the save went through, so loaded the save up, was the old save, tried to load the savestate, but hit the save instead of load keys.... net loss, just over 3 hours of gameplay...

Most of it was just training up my last 3 characters magic before tackling the floating continent (everyone but Celes, Relm, and Strago have every spell learnable in WoB learned already), so not much storyline loss, last save was right before meeting the Espers in Esper Cave, so maybe 20 minutes to get back to where I was in the storyline, but still....fuck. (Did I mention my OCD... Yea, I'm a completionist, lol)

Fuck it, I'll just train them up in WoR, some of those Espers teach the skills faster, and there are better spots to get mass amounts of magic points... But tha5t completely killed my will to play anymore tonight. I'll probably break into the WoR stuff for testing tomorrow.

**Comment by edale2** [comment]
[ff3-74v2.zip](https://github.com/Insidious611/DancingMadFF6/files/1439952/ff3-74v2.zip)
Final version for the house burning down effect.

I searched far and wide, across a few dozen sound clip and video sites to find a good track to use for the house burning down. No matter what I found (and there are A LOT of fire sound clips out there), even clips of sounds from actual house and forest fires (usually because they were recorded from too far away), matched what I felt that scene needed.

I finally narrowed it down to 4 clips, converted them to PCM, and tested them out. First one (which I was really hopeful for) didn't work, because there were all sorts of pops and crackles that weren't the type I was looking for (clipping), lol.  Second one just didn't sound enough like a real fire through the game.

Third one was a 15 second clip, had a perfect crackling wood sound I was looking for (which is ironic, because it wasn't wood burning), but was just a steady chunk of crackling, without much variation, so it wounded too artificial for something like a house burning down.

Fourth track was a 30 second clip, it was one from an actual house fire, , and even had the sound of a section of the house collapsing, but was far enough away to not make being _in_ a burning house believable.

And that's when genius struck!

I copied the full loop of the 15 second clip, and pasted it to the end of the clip, to bring it up to 30 seconds, then I imported the 30 second clip, and rendered them together into one track. Found a new suitable loop in the new track, and converted it to PCM.

When I tested it out, I knew I had a winner. I've been in a house fire before, and this is what it sounds like. once it really gets going

**Comment by edale2** [comment]
OK:
Track 30 is good with the "OST" track.
Track 58 is good with the current version.
Track 74 is good with the v2 version I just posted.

These three tracks are good for release with the current patch.

---

Track 57 is good everywhere except the Rich Man's House in South Figaro.

Once you figure out the coding to fix #38, I'm hoping that'll allow you to also fix the coding for this bit. I'd actually recommend this bit play the SPC audio for track 57 if you get this working (probably easier than making sure the volume adjusts properly for the PCM anyways).

**Comment by Dizzy611** [comment]
Waiting on whether the current experimental fix for issue #38 works for track 57 before uploading the ambient sounds as a batch and closing this issue.

**Comment by edale2** [comment]
I had a thought about the issue with track 57.

For things like exiting the menu and changing rooms, you have checks for what track is playing and where in the track that is (for resume and such). But, do you have a check on the volume it's playing at, so it continues at the same volume?

**Comment by Dizzy611** [comment]
Actually my code isn't that smart. It basically piggybacks on FF3's builtin checks, I don't do any checks for any of the things you mentioned on my own. The MSU-1 hardware itself (or obviously, the routine handling it in your emulator) handles keeping track of song position for resume, I just tell it to pause when necessary and then all I have to do is load the same track again and it resumes from there. I keep track of what song is playing by essentially messing with FF3's own song-playing RAM address, with some minor exceptions which are why the "MSULastTrackSet" address exists. This is done to keep my own code and RAM footprint to a minimum.

For the volume thing, I actually don't keep track of volumes at all atm. I essentially discard them after setting the MSU to them, and FF3's routines are made in most cases to think the SPC is silent.

Obviously if I ever get volume fading working I would need to keep track of the current volume of the MSU-1 for that.

**Comment by Dizzy611** [comment]
Officially marking this as "postponed" as the remaining ambient sounds need fading to work properly.

**Comment by Dizzy611** [comment]
Proper work on this has resumed. Track 57 should now be fixed in master, other tracks I'm still testing to make sure they work right. Also trying to get Nighty Night (the inn jingle) going, using the Pixel Remaster one.
